### PR TITLE
BaSP-TG-106-FE: Some Fixes

### PR DESCRIPTION
**Pull request by Johsua30**
These fixes use changes from TG-108 in the backend.

- Fixed Workedhours in Tasks saving as a number instead of string. Now workedhours should save and load without issues.
- Changed Status in Tasks to Pending, Done and Unfinished. This replaces the old Status.
- Added in all forms the prop shouldFocusError: false. This will make it so on submit the page won't focus on the first field with an error.
- Project and Task names can now contain spaces.
- Added a .pattern(regex) to descriptions so it must at least contain a word before any other simbols.

### CODE REVIEW

File: src/Components/Admin/Tasks/Form/Form.jsx [hunk 1]
```diff
@@ -32,24 +32,25 @@ const taskSchema = Joi.object({
...
-      'string.min': 'Invalid status, it must contain more than 2 letters',
-      'any.only':
-        'Invalid status, it must be one of the following: To do, In progress, Review, Blocked, Done, Cancelled',
-      'string.empty': 'Status is a required field'
-    })
+      'string.min': 'Invalid description, it must contain more than 6 letters',
+      'string.max': 'Invalid description, it must not contain more than 150 letters',
+      'string.empty': 'Description is a required field',
+      'string.pattern.base': 'Task description must contain at least one word'
+    }),
```
Comment (iaracriscenti) [inline src/Components/Admin/Tasks/Form/Form.jsx:48]: Same here

File: src/Components/Admin/Projects/Form/index.js [hunk 2]
```diff
@@ -21,9 +21,11 @@ const schema = Joi.object({
       'string.empty': 'Project name is a required field'
     }),
   description: Joi.string()
+    .min(4)
     .pattern(/(.*[a-zA-Z]){4}/)
     .required()
     .messages({
+      'string.min': 'Description must contain more than 4 letters',
```
Comment (iaracriscenti) [inline src/Components/Admin/Projects/Form/index.js:28]: I suggest changing the min message to:
```
'string.min': 'Description must contain more than 4 characters'
```

File: src/Components/Admin/Tasks/Edit/Edit.jsx [hunk 3]
```diff
@@ -24,27 +24,33 @@ const taskSchema = Joi.object({
...
+    .min(6)
+    .max(150)
+    .required()
+    .pattern(/(.*[a-zA-Z]){4}/)
+    .messages({
+      'string.min': 'Invalid description, it must contain more than 6 letters',
+      'string.max': 'Invalid description, it must not contain more than 150 letters',
+      'string.empty': 'Description is a required field',
+      'string.pattern.base': 'Task description must contain at least one word'
+    }),
```
Comment (iaracriscenti) [inline src/Components/Admin/Tasks/Edit/Edit.jsx:48]: I suggest changing the min, max and pattern messages to:
```
'string.min': 'Invalid description, it must contain more than 6 characters'
'string.max': 'Invalid description, 'Invalid description, it must not contain more than 150 characters'
'string.pattern.base': 'Task description must contain at least 4 letters'
```

**Review by iaracriscenti — APPROVED** [review]
I would modify some messages but all works correctly!

**Review by agusting19 — APPROVED** [review]
Nice work! Also really liked your ticket / pr description

**Review by NicolasFontana — APPROVED** [review]
Everything looks nice!

**Review by tinchorova — APPROVED** [review]
Good job! 😄

**Pull request merged**